server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java

```diff
@@ -1111,7 +1111,7 @@ private UserVm importVirtualMachineInternal(final UnmanagedInstanceTO unmanagedI
             }
         }
         allDetails.put(VmDetailConstants.ROOT_DISK_CONTROLLER, rootDisk.getController());
-        if (cluster.getHypervisorType() != Hypervisor.HypervisorType.VMware) {
+        if (cluster.getHypervisorType() == Hypervisor.HypervisorType.KVM && isImportUnmanagedFromSameHypervisor) {
             allDetails.put(VmDetailConstants.ROOT_DISK_SIZE, String.valueOf(rootDisk.getCapacity() / Resource.ResourceType.bytesToGiB));
```

Comment:
@nvazquez I think we need to here make sure the minimum is 1. i.e.
```suggestion
            long size = rootDisk.getCapacity() / Resource.ResourceType.bytesToGiB;
            if (size <= 0) {
                size = 1;
            }
            allDetails.put(VmDetailConstants.ROOT_DISK_SIZE, String.valueOf(size));
```